### PR TITLE
Add occ command to get app path

### PR DESCRIPTION
**Pull request by VicDeo**
Needed to upgrade the app

### CODE REVIEW

File: core/command/config/app/getpath.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class GetPath extends Base {
+	protected function configure() {
+		parent::configure();
+
+		$this
+			->setName('config:app:getpath')
```
Comment (nickvergessen) [inline core/command/config/app/getpath.php]: `app:getpath` only, this has nothing to do with `\OC\Appconfig`

File: core/command/config/app/getpath.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	 * Executes the current command.
+	 *
+	 * @param InputInterface  $input  An InputInterface instance
+	 * @param OutputInterface $output An OutputInterface instance
+	 * @return null|int null or 0 if everything went fine, or an error code
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		$appName = $input->getArgument('app');
+		$path = [
+			'path' => strval(\OC_App::getAppPath($appName))
```
Comment (nickvergessen) [inline core/command/config/app/getpath.php]: you only have one value, so I'd say
```php
$appName = $input->getArgument('app');
$path = \OC_App::getAppPath($appName);
if ($path !== false) {
	$output->writeln(\OC_App::getAppPath($appName));
	return 0;
}

// App not found, exit with non-zero
return 1;
```

File: core/command/config/app/getpath.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OC\Core\Command\Config\App;
```
Comment (nickvergessen) [inline core/command/config/app/getpath.php]: Also no config here,

**Comment by VicDeo** [comment]
@nickvergessen thanks, updated

File: core/register_command.php [hunk 4]
```diff
@@ -53,6 +53,7 @@
 	$application->add(new OC\Core\Command\Config\App\DeleteConfig(\OC::$server->getConfig()));
 	$application->add(new OC\Core\Command\Config\App\GetConfig(\OC::$server->getConfig()));
 	$application->add(new OC\Core\Command\Config\App\SetConfig(\OC::$server->getConfig()));
+	$application->add(new OC\Core\Command\App\GetPath());
```
Comment (nickvergessen) [inline core/register_command.php]: Does this need an installed owncloud, or can it work without (like app:check-code)?
if so, please move out of `if (\OC::$server->getConfig()->getSystemValue('installed', false)) {`

+ alphabetic order?

Comment (VicDeo) [inline core/register_command.php]: @nickvergessen I don\t see a reason for it to work if OC is not installed. Order has been fixed.

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
squashed

**Comment by LukasReschke** [comment]
:+1:

**Pull request merged**